web/models.py

```
@@ -3046,3 +3046,31 @@
 
     def __str__(self):
         return f"Response by {self.user.username} to {self.question.text}"
+
+
+class ClassroomGameState(models.Model):
+    """Model to store virtual classroom game state for each user."""
+    user = models.ForeignKey('auth.User', on_delete=models.CASCADE, related_name='classroom_state')
+    session_id = models.CharField(max_length=100, blank=True, null=True, help_text="Optional classroom session identifier")
```

Ruff (0.8.2)

Avoid using `null=True` on string-based fields such as `CharField`

(DJ001)